### DISCUSSION TITLE
VBET Sausage Links - Metrics wishlist brainstorm

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/riverscapes-tools#108] [VBET]
9-6-21 Making tasks for transfer over to metrics sheet in #395
## Global Metrics
These are reported for every sausage link and then summary statistics are reported in dashboard and report
### Basic

- [ ] Area (hectares, sq km, acres, sk mi.)

- [ ] Landscape Proportion VB (e.g. for a HUC there is 200 km2 of area, and 5% is valley bottom)
  - [ ] Gross
  - [ ] By network/riverscape category (e.g. perennial, intermittent,  i.e. NHD FCODE)
  - [ ] By Ownership
  - [ ] By Ecoregion

- [ ]  Length of transportation network intersecting valley bottom  (or specific - road, road type, railroad)

### RS Context stuff we have
Not sure if this belongs in RCAT or here because it might not require run-specific parametrization
- [ ] Land Use Intensity
- [ ] LANDFIRE Veg Class proportions within - producing a categorical bar plot distribution or pie chart

### Geometry

- [ ] Total Riverscape Length
- [ ] Integrated Riverscape Width
- [ ] Bankfull Width (from Input)
- [ ] Ratio of Bankfull Width to Valley Bottom Width
- [ ] Total Channel Length
- [ ] Primary Channel Length
- [ ] Valley Bottom Sinuosity = RS Length / Straight-line Riverscape Length
- [ ] Sinuosity Flavors (GNAT)?
- [ ]  Effective Flow Length
- [ ]  Riverscape Density (Riverscape Length / Total Watershed Area) 

 

## Interactive Tools

- [ ]  Playing with probabilistic threshold as first-cut estimate of Active Channel vs. Active Floodplain vs. Inactive Floodplain
- [ ]  Playing with detrended elevation threshold as first-cut estimate of Stage 0 surfaces, incised surfaces, Inactive surfaces (almost TErX within VB)
- [ ]  Once symbolized and segregated, give areas (and horizontal bar plots) by above categories - 
  - [ ]  Proportion Active
  - [ ]  Proportion Inactive 
  - [ ]  Proportion Active Channel
  - [ ]  Proportion Active FP
- [ ]  For those polygons, calculate average height above or below Stage 0 gradeline
- [ ]  Maybe more RCAT... but Recovery Potential can be estimated initially and conservatively by:
  - [ ]  A user-defined LAND USE INTENSITY THRESHOLD 
   - [ ]  Floodplain Inaccessibility 

The above could be produced as horizontal bar plots for each of the thresholds

## Mask Metrics
Using the concept we've described in GCD of AOI vs. Class Masks, to produce same global metrics, but for particular areas of interest, or of polygons of same type (e.g. USFS vs. BLM vs. Private) summaries.

## Local (Sausage Link) Metrics
Is actually all above...

**Comment by joewheaton** [comment]
@MattReimer this is where we laid out the Masking ideas.

@lauren-herbine and @shelbysawyer, please have a read and start asking questions if anything isn't clear here.

**Comment by wally-mac** [comment]
@joewheaton thanks for flushing this out.  @Cashe93 and I have been brainstorming about similar metrics for the on-screen digitizing that we have been doing for the BLM and BoR in Utah so it's great to see that we are coming up with similar and the same metrics.

I understand the Metrics you have come up and think that they will add a ton of value **but I'm not sure what you are asking for.  Is it an attribute table in a shapefile that we populate and pass on to NAR, or what?**  

Please let me know and then we'll work on the task.

**Comment by joewheaton** [comment]
Just a reminder to folks, please provide some critical feedback here. @lauren-herbine and @shelbysawyer I will ask you to provide yours in some vignettes you can link to from here.

**Comment by wally-mac** [comment]
@joewheaton , @lauren-herbine , @shelbysawyer @Cashe93 Here is some feedback on the Global Metrics:

**Geometry** 
- 	In order to have the lengths calculated we will need to run a centerline of the valley bottom and the centerline will house the global metric attributes. 

**Interactive Tools**

_Active/inactive floodplain comparisons_

- As we know, currently our tools do not generate Active Channel, Active Floodplain, and Inactive Floodplain. So, currently this can only be done where these features have been on-screen digitized.
	Note: As of 12/1/20, Karen and Tyler are working on extending RIM to include these comparisons

_Detrended elevation thresholds_

- 	This seems super useful but we need to discuss this more as a team and get all the specifics on how to do this lined out.  

_Recovery potential_
- To make sure we are on the same page, an initial and conservative Recovery potential polygon could be derived from the accessible portion of the valley bottom, which could be further refined by land use intensity values. Is that what you are thinking?

**Reply by joewheaton** [reply]
Bringing this important and topical discussion to the forefront again. NOW is the crunch time on getting VBET right and to get the team humming the same tunes together. I'm mentioning each of you with some tasks below so you can chime in on this discussion. We'll shape up this into a general strategy and then we'll split out into some tasks we can triage, prioritize and assign. If you are mentioned below, get engage please (read above) and add something here, get on a call with each other, meet, talk.

- [x] @shelbysawyer this is your thesis... these metrics and dashboards are how we "effectively communicate and reach our target audiences". You should make sure you understand each of these things above and if we need to start separate discussions or issues that reference this, for specifics, just do it. Ultimately, you will write some vignettes (gists) on the pieces of these we need to flush out more.
- [x] @philipbaileynar put bounds on this discussion as you see fit. 
- [x] @nick4rivers you will not have seen this thread yet, but you will find it parallels much of how you are already thinking and things I know are on your wishlist over in [RIPT Riverscapes Studio - Desktop QGIS](https://github.com/Riverscapes/RIPTPlugin). Please chime in to this discussion (quote specific things) and the parts of this we should be migrating in to your tool, lets queue that up as enhancements.
- [x] @lauren-herbine so much of this is fundamental to what you need for your River Styles work (both at reach-scale and network models). You have a ton of great vignettes and other issues you've done to flesh these things out. You have done those largely in isolation. I want to bring you into _this_ discussion and get you working with all the mentioned folks here so that they are aware of what you know.
- [x] @wally-mac we just need to bring the @Riverscapes/etal-fhc thinking you and @nbouwes and @sshahverdian have been doing in thinking about riverscapes monitoring, into the fold on this and get people talking the same language.
- [ ] @KellyMWhitehead as the guy most often responsible for implementing these metrics, you need to be shaping these conversations in three more forceful ways: 1) put some sideboards on discussions for folks with what is possible, easier, harder in implementing the stuff folks cook up that they want, 2) making your own suggestions/opinions known to the group on ways forward (you have great ideas too, let us hear them) and 3) let folks know what you actually did (i.e. document it and announce it... see [here for preview](#322 ) of one way to do this).

**Reply by philipbaileynar** [reply]
The next steps are for someone at USU to produce these metrics for a demonstration area. It can be a HUC12, HUC10 or whatever. Someone needs to go and calculate the candidate metrics, review them with the group, then narrow them down to a final set. The VBET in question can be totally incorrect and is not relevant. The goal is to prove how each metric is calculated, that we have the necessary data to calculate it, and that it's a worthwhile metric to produce.

The output of this task are:

* a spreadsheet showcasing the metric values for the demo area.
* a list of final metric definitions:
   * metric "display name" (plain english for UI)
   * metric "machine name" (not as fancy as it sounds. Simple unique name, no spaces, no special characters)
   * list of input datasets
   * calculation method in plain English or pseudo code.

This task goes hand in hand with the VBET UI that @joewheaton is producing. i.e. the final metrics should match those shown in the UI mockup that @joewheaton produces.

**Reply by wally-mac** [reply]
@Cashe93 this is a task that I would like you take the lead on. @Tanmanfalcon (Tanner) and @tyler1218hatch can help you out where needed.  I think we could use the Marys River SQL VBET for the demonstration area. If you have questions please add them to this chain.

Thanks!

**Reply by Cashe93** [reply]
@wally-mac and @philipbaileynar, I think I understand the gist of this task but could use some clarification/specification. For the Marys River, I need to roughly map some tier1 GUs:

-  active/inactive floodplain
-  valley bottom centerline
- active channel
- inundation (ponded, free flow, overlfow)
- thalwegs (primary & secondary) 

Once these have been mapped we need to calculate the metrics that were listed at the [top of this thread](https://github.com/Riverscapes/riverscapes-tools/discussions/108#:~:text=Basic,Sausage%20Link)? Please advise.

**Reply by wally-mac** [reply]
@Cashe93 I want you to do as you described **BUT** the task at hand is something entirely different.  Please re-read what Philip wrote above regarding VBET. @tyler1218hatch can you please help Cashe with understanding the task?

**Reply by Cashe93** [reply]
@wally-mac gotcha. I've just met with Tyler and have a better grasp of what needs to be done. I'll keep you updated on progress.

**Reply by lauren-herbine** [reply]
@Cashe93 @tyler1218hatch would you all mind briefly outlining what you discussed? It would be helpful for @shelbysawyer and I. Thanks!

**Reply by Cashe93** [reply]
@lauren-herbine and @shelbysawyer Sure thing! For my benefit, we just went over what specifically needs to be done for this task and why.  In a nutshell, we will be taking an existing VBET output and: 

1. manually calculating the [metrics](https://github.com/Riverscapes/riverscapes-tools/discussions/108#:~:text=Basic,actually%20all%20above...) that Joe had laid out at the beginning of this discussion thread
2. for each metric, documenting how it was calculated and what inputs/datasets were used to calculate the metric.

This will help us determine the feasibility of calculating each metric as well as help us narrow down which final set of metrics should be reported with each VBET output.  My confusion spawned from some of the discussions that were had between the initial post and now and I just needed a push in the right direction. @tyler1218hatch feel free to add anything I may have missed.

**Reply by Cashe93** [reply]
@lauren-herbine and @shelbysawyer, Tyler and I will be meeting tomorrow morning at 10:30 am to go over how we should be calculating some of these metrics. If you would like to join [here is the zoom link](
 https://usu-edu.zoom.us/j/87235147444?pwd=V0JRbDdCYTVzNHZoMnNoTEZSM20zdz09).

**Comment by shelbysawyer** [comment]
We've talked about users being able to specify their specific project area. When users do this, it seems very unlikely that their custom area would perfectly line up with our pre-determined "sausage" segments. This makes me wonder about the dashboard and its capabilities.. If the idea is to have a dashboard provide a set of metrics for each valley bottom segment (ex. proportion valley bottom by land ownership), are we only able to provide these metrics for the existing segments? What is our ability to do some of these metrics on the fly for custom areas which have either been digitized in the app by the user or uploaded via shapefile, etc? 

In terms of the metrics that VBET will provide.. my thought it that the report will mirror what can be accessed in the app. @lauren-herbine and I discussed the possibility of offering customized reports, meaning allowing users to specify the metrics they care about before downloading. Is it better just to offer a standardized report? What is our capability to do customized?

I have additional questions, but I wanted to get some discussion rolling on these now. I can break these out into separate discussions if needed.

**Reply by lauren-herbine** [reply]
@shelbysawyer [here](https://github.com/Riverscapes/riverscapes-tools/issues/225#issuecomment-843279878) is some info from a discussion @joewheaton, @philipbaileynar, and @KellyMWhitehead and I were having regarding calculating gradient and slope a couple of months ago. This might be relevant to your above comment/our conversation from yesterday. Linked above to original discussion, quoted below:

> Thinking about @philipbaileynar :

> @philipbaileynar let's start segmenting at 500m. I think level IV ecoregions would be too long in some cases and could "pinch" a reach in others:

> and

> The length over which we measure sinuosity affects the result. How do we pick the right length if our reaches are so fragmented that they are nearly all straight?

> So another approach is with @KellyMWhitehead and Carol stole from the Fluvial Corridor Tool, in which some metrics can be calculated at length scales other than their segmentation with a moving window (also facilitates calculating at multiple scales). In other words, it doesn't matter what your segmentation length scale is, you choose a "metric length" calculation scale appropriate for the calculation (e.g. a fixed length window of VB for example). This is the approach laid out in Notebaert & Piegay (2013) of using DGOs and AGOs...
![image](https://user-images.githubusercontent.com/73319684/124156466-e7714400-da54-11eb-9497-5086ed369327.png)


> Notebaert B, Piegay H. 2013. Multi-scale factors controlling the pattern of floodplain width at a network scale: The case of the Rhone basin, France. Geomorphology 200 : 155–171. DOI: 10.1016/j.geomorph.2013.03.014

**Reply by shelbysawyer** [reply]
Thanks @lauren-herbine! This is an interesting approach... @joewheaton do you think this is likely what we're leaning towards for VBET?

**Reply by joewheaton** [reply]
> Thanks @lauren-herbine! This is an interesting approach... @joewheaton do you think this is likely what we're leaning towards for VBET?

Absolutely. This is what we will be doing in QRiS as well. More to follow in #326 and other places. stay tuned.
